.travis/refresh-pot.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+
+# We'll refresh the .pot files only when pushing to this branch
+RPOT_PROCESS_BRANCH='master'
+
+RPOT_REPOSITORY_OWNER='audacity'
+RPOT_REPOSITORY_NAME='audacity'
+RPOT_COMMIT_AUTHOR_NAME='Audacity TravisCI Bot'
+RPOT_COMMIT_AUTHOR_EMAIL='travisci-bot@audacityteam.org'
+RPOT_COMMIT_NAME_BASE='Automatic update of .pot files'
+RPOT_COMMIT_NAME="[skip ci] ${RPOT_COMMIT_NAME_BASE}"
+
+if test "${TRAVIS_PULL_REQUEST:-}" != 'false'; then
+    printf '%s: skipping because it''s a pull request.\n' "${RPOT_COMMIT_NAME_BASE}"
+    exit 0
+fi
+
+if test "${TRAVIS_BRANCH:-}" != "${RPOT_PROCESS_BRANCH}"; then
+    printf '%s: skipping because pushing to "%s" instead of "%s".\n' "${RPOT_COMMIT_NAME_BASE}" "${TRAVIS_BRANCH:-}" "${RPOT_PROCESS_BRANCH}"
+    exit 0
+fi
+
+if test "${TRAVIS_REPO_SLUG:-}" != "${RPOT_REPOSITORY_OWNER}/${RPOT_REPOSITORY_NAME}"; then
+    printf '%s: skipping because repository is "%s" instead of "%s/%s".\n' "${RPOT_COMMIT_NAME_BASE}" "${TRAVIS_REPO_SLUG:-}" "${RPOT_REPOSITORY_OWNER}" "${RPOT_REPOSITORY_NAME}"
+    exit 0
+fi
+
+if test -z "${GITHUB_ACCESS_TOKEN:-}"; then
+    printf '%s: skipping because GITHUB_ACCESS_TOKEN is not available
+To create it:
+ - go to https://github.com/settings/tokens/new
+ - create a new token
+ - sudo apt update
+ - sudo apt install -y build-essential ruby ruby-dev
+ - sudo gem install travis
+ - travis encrypt --repo %s/%s GITHUB_ACCESS_TOKEN=<YOUR_ACCESS_TOKEN>
+ - Add to the env setting of:
+   secure: "encrypted string"
+' "${RPOT_COMMIT_NAME_BASE}" "${RPOT_REPOSITORY_OWNER}" "${RPOT_REPOSITORY_NAME}"
+    exit 0
+fi
+
+printf '%s: checking out %s\n' "${RPOT_COMMIT_NAME_BASE}" "${TRAVIS_BRANCH}"
+git checkout -qf "${TRAVIS_BRANCH}"
+
+RPOT_TMPDIR=$(mktemp -d)
+cleanup () {
+    rm -rf "$RPOT_TMPDIR" || true
+}
+trap cleanup EXIT
+
+printf '%s: updating languages\n' "${RPOT_COMMIT_NAME_BASE}"
+cd "${TRAVIS_BUILD_DIR}/locale"
+grep -v -E '^"POT-Creation-Date: [0-9]{4}-[0-9]{2}-[0-9]{2}' audacity.pot > "${RPOT_TMPDIR}/audacity.pot-pre"
+./update_po_files.sh
+grep -v -E '^"POT-Creation-Date: [0-9]{4}-[0-9]{2}-[0-9]{2}' audacity.pot > "${RPOT_TMPDIR}/audacity.pot-post"
+
+printf '%s: checking changes\n' "${RPOT_COMMIT_NAME_BASE}"
+RPOT_CHANGES_DETECTED=0
+if test -n "$(diff "${RPOT_TMPDIR}/audacity.pot-pre" "${RPOT_TMPDIR}/audacity.pot-pre")"; then
+    printf -- '- changes detected in audacity.pot\n'
+    git add --all "${TRAVIS_BUILD_DIR}/locale/audacity.pot"
+    RPOT_CHANGES_DETECTED=1
+fi
+
+if test ${RPOT_CHANGES_DETECTED} -eq 0; then
+    printf '%s: skipping because assets are already up-to-date\n' "${RPOT_COMMIT_NAME_BASE}"
+    exit 0
+fi
```

Comment:
Since the commit starts with `[skip ci]` (see [here](https://github.com/audacity/audacity/pull/335/commits/0c2f866a2819c01f1ce82ffcf556033c1c3dd2c1#diff-84546c93e055bf16c01893c20f0b794fR14) and [here](https://github.com/audacity/audacity/pull/335/commits/0c2f866a2819c01f1ce82ffcf556033c1c3dd2c1#diff-84546c93e055bf16c01893c20f0b794fR77)), the push by the TravisCI job won't trigger another execution of the TravisCI jobs (see [here](https://docs.travis-ci.com/user/customizing-the-build/#skipping-a-build)).

The `POT-Creation-Date` is always updated, even if there are no changes.
See this sample session for example:

```sh
$ cd locale
$ ./update_po_files.sh >/dev/null 2>/dev/null
$ cp audacity.pot audacity.pot.1
$ ./update_po_files.sh >/dev/null 2>/dev/null
$ diff -u audacity.pot audacity.pot.1
--- audacity.pot        2019-01-24 19:19:48.482864400 +0100
+++ audacity.pot.1      2019-01-24 19:19:41.700757400 +0100
@@ -8,7 +8,7 @@
 msgstr ""
 "Project-Id-Version: audacity 2.3.0\n"
 "Report-Msgid-Bugs-To: audacity-translation@lists.sourceforge.net\n"
-"POT-Creation-Date: 2019-01-24 19:19+0100\n"
+"POT-Creation-Date: 2019-01-24 19:18+0100\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
```

Reply:
Ah, the other thing I misunderstood was the grep command; `grep -v -E '^"POT-Creation-Date: [0-9]{4}-[0-9]{2}-[0-9]{2}'` means (due to the `-v`) extract all lines that _don't_ include the creation date, so only if the other content changed would the commit happen.  OK, this makes sense.